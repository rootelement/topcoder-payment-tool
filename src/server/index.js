/**
 * Server initialization.
 */

import Application from 'shared';
import { factory as reducerFactory } from 'reducers';
import { redux, server as serverFactory } from 'topcoder-react-utils';

import webpackConfigFactory from '../../webpack.config';

const mode = process.env.BABEL_ENV;

async function beforeRender(req) {
  const store = await redux.storeFactory({
    getReducerFactory: () => reducerFactory,
    httpRequest: req,
  });
  return { store };
}

global.KEEP_BUILD_INFO = true;
serverFactory(webpackConfigFactory(mode), {
  Application,
  beforeRender,
  devMode: mode === 'development',
});
