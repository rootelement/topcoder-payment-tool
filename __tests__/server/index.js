jest.mock('shared', () => 'APPLICATION');
jest.mock('reducers', () => ({ factory: jest.fn(() => () => ({})) }));
jest.mock('../../webpack.config', () => () => ({}));

jest.mock('topcoder-react-utils', () => {
  const TRU = require.requireActual('topcoder-react-utils');
  return {
    ...TRU,
    server: jest.fn(),
  };
});

test('Passes basic tests', () => {
  require('server');
  const { server } = require('topcoder-react-utils');
  expect(server.mock.calls).toHaveLength(1);
  expect(server.mock.calls[0]).toMatchSnapshot();
  const { beforeRender } = server.mock.calls[0][1];
  expect(() => beforeRender()).not.toThrow();
});
