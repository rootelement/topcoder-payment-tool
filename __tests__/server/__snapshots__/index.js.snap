// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes basic tests 1`] = `
Array [
  Object {},
  Object {
    "Application": "APPLICATION",
    "beforeRender": [Function],
    "devMode": false,
  },
]
`;
